src/test/java/tennisclub/dao/RankingDaoImplTest.java

```diff
@@ -0,0 +1,215 @@
+package tennisclub.dao;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.test.annotation.DirtiesContext;
+import tennisclub.entity.Court;
+import tennisclub.entity.Tournament;
+import tennisclub.entity.User;
+import tennisclub.entity.ranking.Ranking;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.transaction.Transactional;
+import javax.validation.ConstraintViolationException;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@Transactional
+@SpringBootTest
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+class RankingDaoImplTest {
+    private Tournament tournament;
+    private Tournament otherTournament;
+    private User user;
+    private User otherUser;
+
+    @Autowired
+    RankingDao rankingDao;
+
+    @PersistenceContext
+    EntityManager manager;
+
+    @BeforeEach
```

Comment:
I don't think we should use BeforeEach annotation, as this invokes the code before each test execution _(we want BeforeClass annotation from TestNG, because it executes only once before the tests are executed)_.

Reply:
@Fuyunori 

> as this invokes the code before each test execution

But that is exactly what we want. Thanks to the `@DirtiesContext(...)` annotation :top:, the whole DB is cleared before each test-case and therefore also needs to be re-filled before each test-case. This behavior is desirable, because then each test-case is completely isolated from the others -- the DB behaves like a "local variable". What you are proposing, would make the tests highly inter-dependent; their order of execution would probably have to be maintained and adding/removing/inserting new tests could break others.

Reply:
Thanks for explaining 😊